app/uk/gov/hmrc/tradergoodsprofilesdatastore/connectors/CustomsDataStoreConnector.scala

```diff
@@ -49,4 +53,17 @@ class CustomsDataStoreConnector @Inject() (config: Configuration, httpClient: Ht
           case _         => Future.failed(UpstreamErrorResponse(response.body, response.status))
         }
       }
+
+  def getEoriHistory(
+    eori: String
+  )(implicit hc: HeaderCarrier): Future[Option[EoriHistoryResponse]] =
+    httpClient
+      .get(eoriHistoryUrl(eori))
+      .execute[HttpResponse]
+      .flatMap { response =>
+        response.status match {
+          case OK => Future.successful(Some(response.json.as[EoriHistoryResponse]))
```

Comment:
I'm not sure that this should be an option - as if there aren't any shouldn't it just be an empty sequence?

Reply:
Updated